book/src/rfcs/benchmarking.md

```diff
@@ -0,0 +1,109 @@
+It can be subdivided it 2 other metrics or alternative approaches:
+
+- Execution time: The most simple, run the program in question several times and take the average, it may become quite expensive if the program measured takes some time because that time will be multiplied by the number of samples, in those cases it can be made faster sacrificing precision. This benchmark is one of the most interesting for external users, and one of the main comparison point for evaluating the competition. It can also serve for internal developer as a regression test or to verify performance improvements.
+
+- CPU instructions: This method instead of recording execution time counts the number of cpu instructions used, not very useful for external developer given that it doesn't translate directly to time. It is mostly useful to internal developer as an alternative to execution time, it is faster given that you only have to run it once to collect the information and can be run on cheap machines without significative noise given that execution time doesn't affect the number of instructions. Additionally the tools used to measure it also collect some other metrics that can be useful to aide optimization like cache misses and branch mispredictions.
+
+#### Memory
+
+Memory usage is another interesting metric, there are different ways of measuring memory but the most useful for us is peak memory, peak memory will many times define if something can even be used in a particular environment. It can also help finding errors related to unusual memory usage.
+It is easier to measure, as long as there is enough memory, noise shouldn't affect the results.
+
+#### Hardware utilization
+
+It can be of interest in some cases to measure how much of the available resources are actually used, ideally the same computer will take half the time with twice the resources, unfortunately this isn't always the case and sometimes additional resources go unused.
+This doesn't change that often, the main utility of these benchmarks is just to detect low utilization so that it can be improved.
+Hardware utilization could include many things, GPU acceleration for example if we were to start using it, but for now will only include CPU/core utilization.
+
+### Layers
+
+There are 3 main layers of the stack that we want to consider, kimchi, snarky and pickles. Given that they are implemented in different languages and exist in different repositories special considerations each layer will be required.
+
+## Where to run the benchmarks
+
+To start everyone with the resources should be able to run the benchmarks locally, additionally we may want some of the benchmarks to run in CI to detect regressions. Some benchmarks may be considerably expensive to run, a decision will have to be made about which benchmarks to run and how often we should run them.
```

Comment:
Detecting regression in performance might also include thinking more about CI infrastructure than other benchmarks. Both for very short or very long tests.